ddtrace/appsec/_iast/_taint_tracking/aspects.py

```diff
@@ -158,16 +169,19 @@ def slice_aspect(candidate_text, start, stop, step) -> Any:
 
 
 def bytearray_extend_aspect(orig_function, flag_added_args, *args, **kwargs):
-    # type: (Callable, int, Any, Any) -> Any
-    if not isinstance(orig_function, BuiltinFunctionType):
+    # type: (Optional[Callable], int, Any, Any) -> Any
+    if orig_function and not isinstance(orig_function, BuiltinFunctionType):
         if flag_added_args > 0:
             args = args[flag_added_args:]
         return orig_function(*args, **kwargs)
 
+    if len(args) < 2:
+        return args[0].extend(*args[1:], **kwargs)
```

Comment:
Could you please add a comment and/or example that explains why we need this logic? If not, we may forget the rationale behind it in a month :sweat_smile:

Reply:
```suggestion
        # If we're not receiving at least 2 arguments, means the call was
        # ``x.extend()`` and not ``x.extend(y)``
        # so either not the extend we're looking for, or no changes in taint ranges.
        return args[0].extend(*args[1:], **kwargs)
```